pandera/backends/pyspark/container.py

```diff
@@ -283,7 +283,7 @@ def collect_schema_components(
         schema_components = []
         for col_name, column in schema.columns.items():
             if (
-                column.required or col_name in check_obj
+                column.required or col_name in check_obj.columns
```

Comment:
A screenshot of the `pdb` session where the `col_name` and `check_obj` are displayed:
<img width="1511" alt="image" src="https://github.com/unionai-oss/pandera/assets/110418479/f7465ad5-49ee-41bd-96ae-da93ac02deb0">

Reply:
Hi @filipeo2-mck. Looking at the debugger log, the change make total sense. However, I assume this is rather a bugfix than adding supprot for optional fields? I.e this is how it should have been implemented from the start, but wasn't properly covered with the right test cases I assume?

Reply:
Thanks for clearifying. That's what I thought!